### ISSUE TITLE
Catastrophic backtracking?

### ISSUE THREAD

**Issue by robertklep** [skerit/epinfer#3]
The following code takes about 3.5 seconds on my MBP:

```
const epinfer = require('epinfer');
const text    = 'abcdefghijklmnopqrstuvwxyzabcdefghijkl.';

console.time('parse');
epinfer.parse(text);
console.timeEnd('parse');
```

Adding an additional letter to the alphabet almost doubles the run time, which suggests catastrophic backtracking.

I'll try and find out which property is causing this.

**EDIT 1**: oh, that was easy. It's related to these:
```
website.addExtract(/(?=[A-Z0-9]+\.)((?:[A-Z0-9]{2,})+\.org)/gi, 10);        // "bla.bla.org"
website.addExtract(/(?=[A-Z0-9]+\.)((?:[A-Z0-9]{2,})+\.com)/gi, 10);        // "bla.bla.com"
website.addExtract(/(?=[A-Z0-9]+\.)((?:[A-Z0-9]{2,})+\.ru)/gi, 10);         // "bla.bla.ru"
website.addExtract(/(?=[A-Z0-9]+\.)((?:[A-Z0-9]{2,})+\.tv)/gi, 10);         // "bla.bla.tv"
```

**EDIT 2**: I'm not quite sure what they are supposed to match. It seems like this would be sufficient (although I still don't understand the point of the `(?=...)` sub-patterns):

```
website.addExtract(/(?=[A-Z0-9]+\.)([A-Z0-9]{2,}\.org)/gi, 10);        // "bla.bla.org"
website.addExtract(/(?=[A-Z0-9]+\.)([A-Z0-9]{2,}\.com)/gi, 10);        // "bla.bla.com"
website.addExtract(/(?=[A-Z0-9]+\.)([A-Z0-9]{2,}\.ru)/gi, 10);         // "bla.bla.ru"
website.addExtract(/(?=[A-Z0-9]+\.)([A-Z0-9]{2,}\.tv)/gi, 10);         // "bla.bla.tv"
```

Tests still pass and the backtracking is fixed.

**Comment by skerit** [comment]
Oh, I'll take a look

**Comment by skerit** [comment]
Should be fixed in commit 638a416789a64950abc05759389dc95709088792

The + shouldn't have been there, they were meant only for the tests above, for the websites delimited by [] brackets.

**Comment by robertklep** [comment]
Thanks!
